great_expectations/data_context/store/datasource_store.py

```diff
@@ -218,3 +222,12 @@ def _determine_datasource_key(self, datasource_name: str) -> DataContextVariable
             resource_name=datasource_name,
         )
         return datasource_key
+
+    def _validate_key(self, key: Union[GeCloudIdentifier, Tuple[str, ...]]) -> None:  # type: ignore[override]
+        # Continue to support tuple as a key for get method until we can refactor
+        if not isinstance(key, GeCloudIdentifier):
+            super()._validate_key(key)  # type: ignore[arg-type]
+        else:
+            assert isinstance(
+                key, GeCloudIdentifier
+            ), f"Only {GeCloudIdentifier.__name__} is supported in {self.__class__.__name__}"
```

Comment:
Nit - don't think we need `if/else` here
```suggestion
    def _validate_key(self, key: Union[GeCloudIdentifier, Tuple[str, ...]]) -> None:  # type: ignore[override]
        # Continue to support tuple as a key for get method until we can refactor
        assert isinstance(
                key, GeCloudIdentifier
            ), f"Only {GeCloudIdentifier.__name__} is supported in {self.__class__.__name__}"
        super()._validate_key(key)  # type: ignore[arg-type]
```

Reply:
I think we do, since we don't want to do the assertion if the key is a tuple. Or am I missing something?